### PR TITLE
EuroSAT: fix SSL certificate issues

### PR DESCRIPTION
**Pull request by adamjstewart**
When downloading the EuroSAT dataset, users encounter the following error message:
```pycon
> python3
>>> from torchvision.datasets import EuroSAT
>>> EuroSAT('data', download=True)
...
urllib.error.URLError: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1006)>
```
This leads a lot of users to dangerous [StackOverflow suggestions](https://stackoverflow.com/q/71263622/5828163) to disable SSL verification entirely.

This PR switches the download URL to a stable mirror of the download on Hugging Face. This dataset was rehosted with permission from the original author of EuroSAT: https://github.com/phelber/EuroSAT/issues/10. The zip file is unmodified and has the same MD5 and SHA256 checksums.

Also added a link to a data loader for the multispectral (MS) version of the dataset for any other remote sensing researchers who need it.

### CODE REVIEW

File: torchvision/datasets/eurosat.py [hunk 1]
```diff
@@ -53,7 +56,7 @@ def download(self) -> None:
 
         os.makedirs(self._base_folder, exist_ok=True)
         download_and_extract_archive(
-            "https://madm.dfki.de/files/sentinel/EuroSAT.zip",
+            "https://huggingface.co/datasets/torchgeo/eurosat/resolve/c877bcd43f099cd0196738f714544e355477f3fd/EuroSAT.zip",
```
Comment (adamjstewart) [inline torchvision/datasets/eurosat.py:59]: Could use `hf.co` or `main` to make this URL shorter if you want. I don't plan on modifying this file, but the commit hash ensures that even if I do, the checksum still works.

**Review by NicolasHug — APPROVED** [review]
Thanks a lot @adamjstewart

**Comment by adamjstewart** [comment]
Are the prototype datasets being abandoned?

**Comment by NicolasHug** [comment]
Yes, pretty much (https://github.com/pytorch/vision/tree/main/torchvision/prototype/datasets).
The underlying datapipes are being deprecated as well https://github.com/pytorch/data/issues/1196#issuecomment-2221200178

**Pull request merged**